src/components/settings/PersonalizedPushSettings.tsx

```diff
@@ -70,41 +80,52 @@ export const PersonalizedPushSettings = () => {
 
   useEffect(() => {
     (async () => {
-      if (pushToken) {
-        const categories = await getExcludedCategoriesPushTokenFromServer(pushToken);
-        setExcludeCategoryIds(categories?.length ? categories : []);
-      }
+      if (!pushToken) return;
+      const categories = await getExcludedCategoriesPushTokenFromServer(pushToken);
+      setExcludeCategoryIds(categories);
     })();
   }, [pushToken]);
 
   useEffect(() => {
-    if (pushToken) {
-      const tagList = queryVariables?.tagList;
+    if (!!Object.keys(excludeCategoryIds).length) {
```

Comment:
`!!` are not needed here, the number returned by `length` is interpreted correctly by the `if` already